source/views/streaming/streams/list.tsx

```diff
@@ -23,99 +24,137 @@ const styles = StyleSheet.create({
+		let enabledCategories = filters.flatMap((f: ListType) =>
+			f.spec.selected.flatMap((s) => s.title),
+		)
+
+		if (enabledCategories.length === 0) {
+			return entries
 		}
-	}, [])
 
-	let refresh = async (): Promise<void> => {
-		setRefreshing(true)
-		await getStreams(true)
-		setRefreshing(false)
-	}
+		return enabledCategories.includes(stream.category)
+	})
+}
 
-	let getStreams = async (
-		reload?: boolean,
-		date: Moment = moment.tz(timezone()),
-	) => {
-		let dateFrom = date.format('YYYY-MM-DD')
-		let dateTo = date.clone().add(2, 'month').format('YYYY-MM-DD')
-
-		let data = await fetch(API('/streams/upcoming'), {
-			searchParams: {
-				sort: 'ascending',
-				dateFrom,
-				dateTo,
-			},
-			delay: reload ? 500 : 0,
-		}).json<Array<StreamType>>()
-
-		data = data
-			.filter((stream) => stream.category !== 'athletics')
-			.map((stream) => {
-				let date: Moment = moment(stream.starttime)
-				let dateGroup = date.format('dddd, MMMM Do')
-
-				let group = stream.status.toLowerCase() !== 'live' ? dateGroup : 'Live'
-
-				return {
-					...stream,
-					// force title-case on the stream types, to prevent not-actually-duplicate headings
-					category: titleCase(stream.category),
-					date: date,
-					$groupBy: group,
-				}
-			})
-
-		let grouped = groupBy(data, (j) => j.$groupBy)
-		let mapped = toPairs(grouped).map(([title, data]) => ({title, data}))
-
-		setStreams(mapped)
-	}
+let useStreams = (date: Moment = moment.tz(timezone())) => {
+	let dateFrom = date.format('YYYY-MM-DD')
+	let dateTo = date.clone().add(2, 'month').format('YYYY-MM-DD')
+
+	return useFetch<StreamType[]>(
+		API('/streams/upcoming', {
+			sort: 'ascending',
+			dateFrom,
+			dateTo,
+		}),
+		{
+			headers: {accept: 'application/json'},
+		},
+	)
+}
 
-	let keyExtractor = (item: StreamType) => item.eid
+export const StreamListView = (): JSX.Element => {
+	let {data = [], error, reload, isPending, isInitial, isLoading} = useStreams()
 
-	let renderItem = ({item}: {item: StreamType}) => <StreamRow stream={item} />
+	let [filters, setFilters] = React.useState<ListType[]>([])
 
-	if (loading) {
-		return <LoadingView />
-	}
+	let entries = React.useMemo(() => {
+		return data.map((stream) => {
+			let date: Moment = moment(stream.starttime)
+			let dateGroup = date.format('dddd, MMMM Do')
+
+			let group = stream.status.toLowerCase() !== 'live' ? dateGroup : 'Live'
+
+			return {
+				...stream,
+				// force title-case on the stream types, to prevent not-actually-duplicate headings
+				category: titleCase(stream.category),
+				date: date,
+				$groupBy: group,
+			}
+		})
+	}, [data])
+
+	React.useEffect(() => {
+		let allCategories = data.flatMap((stream) => titleCase(stream.category))
+
+		if (allCategories.length === 0) {
+			return
+		}
+
+		let categories = [...new Set(allCategories)].sort()
+		let filterCategories = categories.map((c) => {
+			return {title: c}
+		})
+
+		let streamFilters: ListType[] = [
```

Comment:
It is possible to tighten `ListType` or rename it to something like `FilterType`? It seems like we're using a type called `List` for a ... list of filters? Shouldn't it just be a list of filters?

Reply:
Are you asking if `ListType` can be `ListFilterType`? It probably could. This is coming from `modules/filter/types.ts` where we define a whole scope of different `FilterType` definitions.

```ts
export type ToggleType = {
	type: 'toggle'
	key: string
	enabled: boolean
	spec: ToggleSpecType
	apply: ToggleFilterFunctionType
}

export type PickerType = {
	type: 'picker'
	key: string
	enabled: true
	spec: PickerSpecType
	apply: PickerFilterFunctionType
}

export type ListType = {
	type: 'list'
	key: string
	enabled: boolean
	spec: ListSpecType
	apply: ListFilterFunctionType
}

export type FilterType = ToggleType | PickerType | ListType
```